### ISSUE TITLE
Default deaccession agent uniqueness rules are different from other interaction agents, causing IntegrityError on save

### ISSUE THREAD

**Issue by combs-a** [specify/specify7#5014]
**Describe the bug**
Default deaccession agent uniqueness rules allows for multiple of the same agent with the same role, as compared to other interaction agent uniqueness roles not allowing for it. This causes an IntegrityError upon save.

**To Reproduce**

1. Go to Interactions -> Deaccession.
2. Add a Deaccession agent and give it a role.
3. Add a second Deaccession agent and give it the same role as the previous.
4. See that saving is not blocked.
5. Save and see error message.

**Expected behavior**
When there are multiple of the same Deaccession agent with the same roles, it should block saving as it does with other interactions.

**Screenshots**

https://github.com/specify/specify7/assets/71564869/d2880cd3-4888-4ef5-83c6-3efa22f3e5de

**Crash Report**
[Specify 7 Crash Report - 2024-06-18T14_34_32.268Z.txt](https://github.com/user-attachments/files/15888133/Specify.7.Crash.Report.-.2024-06-18T14_34_32.268Z.txt)

Please, also fill out the following information manually:

- OS: Windows 10
- Browser: Firefox
- Specify 7 Version: `edge`
- Database Name: `sdnhm_herps_2_8_24`
- Collection name: `Herp-PC`
- User Name: `testuser`

**Comment by acwhite211** [comment]
I was able to get the error `django.db.utils.IntegrityError: (1062, "Duplicate entry 'Other-1876-8' for key 'Role'")` when I added the same agent twice with the same role "Other".  It did work though when I added two different agents with the same "Other" role.  Is this the desired behavior?

**Comment by combs-a** [comment]
Yeah, that's the correct behavior. It should only be blocked when it's multiple of the same agent with the same role.

**Comment by emenslin** [comment]
Can recreate in edge (7.9.6)
